### PR TITLE
Add Storybook GitHub workflow for publishing to GH pages

### LINKED ISSUE #35637: Publish woocommerce/components storybook
**Issue Description:**

After moving WooCommerce Admin to the mono repo, we stopped publishing `woocommerce/components` storybook within Github.io ( [original link](https://woocommerce.github.io/woocommerce-admin/#/components/) ) we should publish this again, but just the Storybook similar to what Gutenberg does.

Likewise with the recent merge of Woo Blocks, we should also include the [WooBlocks story book](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-blocks/storybook) if possible.

I believe this may be the action Gutenberg uses: https://github.com/WordPress/gutenberg/blob/trunk/.github/workflows/storybook-pages.yml

**Acceptance criteria**

- [ ] Have WooCommerce Storybook deployed on a woocommerce.github.io page
- [ ] Create an action that either runs whenever a storybook has been updated ( any of the packages/js folders ) OR one that can easily be run manually OR run on a cron once a week.
- [ ] Woo Blocks is included as part of the Storybook

### PR DESCRIPTION
**Pull request by louwie17**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This is a follow up of https://github.com/woocommerce/woocommerce/pull/51168 and will publish the WooCommerce storybook to pages. The #51168 PR fixes the storybook build command so this should work as expected now.

Closes https://github.com/woocommerce/woocommerce/issues/35637

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check if `Storybook GitHub Pages` action ran correctly in this PR, it gets published to the `gh-pages` branch within the repo: https://github.com/woocommerce/woocommerce/tree/gh-pages
2. Check out https://woocommerce.github.io/woocommerce to see if Storybook works correctly.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: .github/workflows/storybook-pages.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+on:
+    workflow_dispatch:
+    push:
+        branches:
+            - trunk
+            - add/storybook_github_workflow
+
+permissions:
+    contents: write
```
Comment (louwie17) [inline .github/workflows/storybook-pages.yml:7]: See: https://github.com/peaceiris/actions-gh-pages?tab=readme-ov-file#%EF%B8%8F-first-deployment-with-github_token

File: .github/workflows/storybook-pages.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Storybook GitHub Pages
+
+on:
+    workflow_dispatch:
+    push:
+        branches:
+            - trunk
+            - add/storybook_github_workflow
```
Comment (louwie17) [inline .github/workflows/storybook-pages.yml]: I can remove my branch before merging this.

File: .github/workflows/storybook-pages.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+            - name: Build Storybook
+              run: pnpm --filter='@woocommerce/storybook' build-storybook
+
+            - name: Deploy
+              uses: peaceiris/actions-gh-pages@v4
+              with:
+                  github_token: ${{ secrets.GITHUB_TOKEN }}
+                  publish_dir: ./tools/storybook/storybook-static
+                  force_orphan: true
```
Comment (louwie17) [inline .github/workflows/storybook-pages.yml:34]: This gets deployed to the `gh-pages` branch on WooCommerce: https://github.com/woocommerce/woocommerce/tree/gh-pages

File: .github/workflows/storybook-pages.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    contents: write
+
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        if: ${{ github.repository == 'woocommerce/woocommerce' }}
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v3
```
Comment (gigitux) [inline .github/workflows/storybook-pages.yml]: Let's use the hash commit to ensure that the same version is used

File: .github/workflows/storybook-pages.yml (hunk 3 above)
Comment (gigitux) [reply .github/workflows/storybook-pages.yml:34]: Let's use the hash commit to ensure that the same version is used

File: .github/workflows/storybook-pages.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+name: Storybook GitHub Pages
+
+on:
+    workflow_dispatch:
+    push:
+        branches:
+            - trunk
```
Comment (louwie17) [inline .github/workflows/storybook-pages.yml]: If we wanted to, we can also limit this to directories and only publish when changes are made in the respective plugins/packages that have GH pages.
That is if we think running on every push to `trunk` is to much. Of-course we can also manually trigger it for the time being or on a weekly basis.

**Review by gigitux — APPROVED** [review]
Thanks for working on this! The PR LGTM! I left two small minor comments that it could make sense to address.

Given that Storybook works, I'm wondering if we should add some reference to the it on https://woocommerce.github.io/ and on https://developer.woocommerce.com/docs/: what do you think?

cc @woocommerce/developer-advocacy

File: .github/workflows/storybook-pages.yml (hunk 4 above)
Comment (louwie17) [reply .github/workflows/storybook-pages.yml]: I am happy to move to a hash commit similar to GB, but at the moment we don't follow that format at all within WooCommerce.
I assume this is mostly for security reasons btw?

Comment (gigitux) [reply .github/workflows/storybook-pages.yml]: > I assume this is mostly for security reasons btw?
 
Yes!

Comment (louwie17) [reply .github/workflows/storybook-pages.yml]: Addressed anyway :)

**Comment by louwie17** [comment]
> Given that Storybook works, I'm wondering if we should add some reference to the it on https://woocommerce.github.io/ and on https://developer.woocommerce.com/docs/: what do you think?
> 
> cc @woocommerce/developer-advocacy

Yeah deff update both, we can update the first one by updating https://github.com/woocommerce/woocommerce.github.io

File: .github/workflows/storybook-pages.yml (hunk 4 above)
Comment (louwie17) [reply .github/workflows/storybook-pages.yml]: Updated as part of this commit: https://github.com/woocommerce/woocommerce/pull/51235/commits/9459ecbf86c27173d3610d1c641f5d5d5ebe7e9a

**Review by vedanshujain — APPROVED** [review]

**Pull request merged**